# Global Warming Analysis - Geospatial Data Analysis Project
Weekly challenge: Week 10  
Date: 9/26/2022

## 1. Problem statement / Objective: 
### To understand the global warming thru geospatial data analysis

## 2. Getting the system ready and loading the data

### Loading the packages

In [1]:
# Import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt # for data visualization
import seaborn as sns # for data visualization
import plotly.express as px # for data visualization

from plotly.offline import init_notebook_mode # for data visualization
init_notebook_mode(connected = True)

### Data

For this project we have 5 data files. We will read these files as and when needed.

1. GlobalLandTemperaturesByCity.csv
2. GlobalLandTemperaturesByCountry.csv
3. GlobalLandTemperaturesByMajorCity.csv
4. GlobalLandTemperaturesByState.csv
5. GlobalTemperatures.csv

### Reading the data

In [2]:
file = 'C:/Users/unpat/_Projects/Challenge_AprojectAweek_data_files/Week_10/GlobalLandTemperaturesByCountry.csv'
global_temp_country = pd.read_csv(file)
global_temp_country.head(3)

dt  AverageTemperature  AverageTemperatureUncertainty Country
0  1743-11-01               4.384                          2.294   Åland
1  1743-12-01                 NaN                            NaN   Åland
2  1744-01-01                 NaN                            NaN   Åland

## 3. Understanding and cleaning the data (Data preprocessing)

In this step, we will understand our data and clean it wherever needed. This includes but not limited to meaning of each column, data type of each column, # of rows and columns, statistical summery of the numerical columns,  take care of the null values as well as duplicate values.

In [3]:
# Checking columns of the test data
global_temp_country.columns

Index(['dt', 'AverageTemperature', 'AverageTemperatureUncertainty', 'Country'], dtype='object')

### Description of each column

dt:  
AverageTemperature:  
AverageTemperatureUncertainty:  
Country:

### Data type of each column

In [4]:
global_temp_country.dtypes

dt                                object
AverageTemperature               float64
AverageTemperatureUncertainty    float64
Country                           object
dtype: object

There are two formats of data types:  
* Object:  
* float64:  

We will convert the date data type when needed.

### Shape of the dataset

In [5]:
shape_ = global_temp_country.shape
shape_

(577462, 4)

There are 577462 rows and 4 columns in the dataset global_temp_country.

### Checking the statistical summary of numerical columns

In [6]:
global_temp_country.describe()

AverageTemperature  AverageTemperatureUncertainty
count       544811.000000                  545550.000000
mean            17.193354                       1.019057
std             10.953966                       1.201930
min            -37.658000                       0.052000
25%             10.025000                       0.323000
50%             20.901000                       0.571000
75%             25.814000                       1.206000
max             38.842000                      15.003000

The data seems to be right skewed.

### Checking for the null values

In [7]:
# Count of null values
isna_ = global_temp_country.isna().sum()
isna_

dt                                   0
AverageTemperature               32651
AverageTemperatureUncertainty    31912
Country                              0
dtype: int64

In [8]:
# % of null values for percent_avg_temp
percent_avg_temp = isna_[1] / shape_[0] * 100
percent_avg_temp

5.654224866744478

In [9]:
# % of null values for percent_avg_temp_uncertain
percent_avg_temp_uncertain = isna_[2] / shape_[0] * 100
percent_avg_temp_uncertain

5.5262510779930105

* There are less than 6% of missing rows in two variables (AverageTemperature, AverageTemperatureUncertainty) each.  
* Deleting these rows won't affect too much to our analysis so I will delete these rows before moving forward.  
* AverageTemperature and AverageTemperatureUncertainty are related to each other so I will have to delete rows from only one column.

In [10]:
global_temp_country.dropna(
    axis = 'index',
    how = 'any',
    subset=['AverageTemperature'],
    inplace = True
)

In [11]:
# Checking for null values after deleting rows with null values
global_temp_country.isna().sum()

dt                               0
AverageTemperature               0
AverageTemperatureUncertainty    0
Country                          0
dtype: int64

We now do not have any missing values in our data

### Review of the country column: # of countries, duplicate values

In [12]:
# Checking for # of countries
global_temp_country['Country'].nunique()

242

There are 242 unique countries in our data.

In [13]:
# Checking of there are any duplicate values.
global_temp_country['Country'].unique()

array(['Åland', 'Afghanistan', 'Africa', 'Albania', 'Algeria',
       'American Samoa', 'Andorra', 'Angola', 'Anguilla',
       'Antigua And Barbuda', 'Argentina', 'Armenia', 'Aruba', 'Asia',
       'Australia', 'Austria', 'Azerbaijan', 'Bahamas', 'Bahrain',
       'Baker Island', 'Bangladesh', 'Barbados', 'Belarus', 'Belgium',
       'Belize', 'Benin', 'Bhutan', 'Bolivia',
       'Bonaire, Saint Eustatius And Saba', 'Bosnia And Herzegovina',
       'Botswana', 'Brazil', 'British Virgin Islands', 'Bulgaria',
       'Burkina Faso', 'Burma', 'Burundi', "Côte D'Ivoire", 'Cambodia',
       'Cameroon', 'Canada', 'Cape Verde', 'Cayman Islands',
       'Central African Republic', 'Chad', 'Chile', 'China',
       'Christmas Island', 'Colombia', 'Comoros',
       'Congo (Democratic Republic Of The)', 'Congo', 'Costa Rica',
       'Croatia', 'Cuba', 'Curaçao', 'Cyprus', 'Czech Republic',
       'Denmark (Europe)', 'Denmark', 'Djibouti', 'Dominica',
       'Dominican Republic', 'Ecuador', 'Egypt'

From the above country name array, we can see that there are a few countries with more than one name so we will have to correct these country names.

In [14]:
dict_ = {
    'Congo (Democratic Republic Of The)':'Congo',
    'Denmark (Europe)':'Denmark',
    'France (Europe)':'France',
    'Netherlands (Europe)':'Netherlands',
    'United Kingdom (Europe)':'United Kingdom'
}

In [15]:
global_temp_country['Country'].replace(dict_, inplace = True)

## 4. Exploratory Data Analysis (EDA)

### Calculate the average temperate of each country

In [16]:
avg_temp = global_temp_country.groupby(['Country'])['AverageTemperature'].mean()
avg_temp

Country
Afghanistan       14.045007
Africa            24.074203
Albania           12.610646
Algeria           22.985112
American Samoa    26.611965
                    ...    
Western Sahara    22.319818
Yemen             26.253597
Zambia            21.282956
Zimbabwe          21.117547
Åland              5.291383
Name: AverageTemperature, Length: 237, dtype: float64

In [17]:
# Converting to a dataframe and resetting the index
avg_temp = avg_temp.to_frame().reset_index()
avg_temp

Country  AverageTemperature
0       Afghanistan           14.045007
1            Africa           24.074203
2           Albania           12.610646
3           Algeria           22.985112
4    American Samoa           26.611965
..              ...                 ...
232  Western Sahara           22.319818
233           Yemen           26.253597
234          Zambia           21.282956
235        Zimbabwe           21.117547
236           Åland            5.291383

[237 rows x 2 columns]

In [18]:
# Plotting choropleth map
fig = px.choropleth(
    data_frame=avg_temp,
    locations='Country',
    locationmode='country names',
    color='AverageTemperature',
    title='Choropleth Map of Average Temperature',
)
# fig.update_layout(title='Choropleth Map of Average Temperature')
fig.show()

### Is there a global warming?

To see the effect of global warming, we need to know how the temperature changed over a long period of time. Global warming is a slow process and its effect can only be felt over a long period of time. We have monthly temperature data of many countries in the world. We will analyze this data and visualize the effect of global warming in this section.

#### Loading the 'GlobalTemperatures.csv' data file

In [19]:
file = 'C:/Users/unpat/_Projects/Challenge_AprojectAweek_data_files/Week_10\GlobalTemperatures.csv'
global_temp = pd.read_csv(file)
global_temp.head(5)

dt  LandAverageTemperature  LandAverageTemperatureUncertainty  \
0  1750-01-01                   3.034                              3.574   
1  1750-02-01                   3.083                              3.702   
2  1750-03-01                   5.626                              3.076   
3  1750-04-01                   8.490                              2.451   
4  1750-05-01                  11.573                              2.072   

   LandMaxTemperature  LandMaxTemperatureUncertainty  LandMinTemperature  \
0                 NaN                            NaN                 NaN   
1                 NaN                            NaN                 NaN   
2                 NaN                            NaN                 NaN   
3                 NaN                            NaN                 NaN   
4                 NaN                            NaN                 NaN   

   LandMinTemperatureUncertainty  LandAndOceanAverageTemperature  \
0                            NaN                             NaN   
1                            NaN                             NaN   
2                            NaN                             NaN   
3                            NaN                             NaN   
4                            NaN                             NaN   

   LandAndOceanAverageTemperatureUncertainty  
0                                        NaN  
1                                        NaN  
2                                        NaN  
3                                        NaN  
4                                        NaN

In [20]:
global_temp.tail(5)

dt  LandAverageTemperature  LandAverageTemperatureUncertainty  \
3187  2015-08-01                  14.755                              0.072   
3188  2015-09-01                  12.999                              0.079   
3189  2015-10-01                  10.801                              0.102   
3190  2015-11-01                   7.433                              0.119   
3191  2015-12-01                   5.518                              0.100   

      LandMaxTemperature  LandMaxTemperatureUncertainty  LandMinTemperature  \
3187              20.699                          0.110               9.005   
3188              18.845                          0.088               7.199   
3189              16.450                          0.059               5.232   
3190              12.892                          0.093               2.157   
3191              10.725                          0.154               0.287   

      LandMinTemperatureUncertainty  LandAndOceanAverageTemperature  \
3187                          0.170                          17.589   
3188                          0.229                          17.049   
3189                          0.115                          16.290   
3190                          0.106                          15.252   
3191                          0.099                          14.774   

      LandAndOceanAverageTemperatureUncertainty  
3187                                      0.057  
3188                                      0.058  
3189                                      0.062  
3190                                      0.063  
3191                                      0.062

Date is associated with the global warming. To see the effect of global warming, we have to know the history of temperature (how the temperature varied over time). Our data goes back to a few hundred years so we would be working with the annual temperature variations.

In [21]:
# Concept of separating year from the date
global_temp['dt'][0].split('-')[0]

'1750'

In [22]:
# Function that will fetch year from date
def fetch_year(date):
    return date.split('-')[0]

In [23]:
# Applying fetch_year function to the dt column of the dataframe
global_temp['years'] = global_temp['dt'].apply(fetch_year)

In [24]:
global_temp.head()

dt  LandAverageTemperature  LandAverageTemperatureUncertainty  \
0  1750-01-01                   3.034                              3.574   
1  1750-02-01                   3.083                              3.702   
2  1750-03-01                   5.626                              3.076   
3  1750-04-01                   8.490                              2.451   
4  1750-05-01                  11.573                              2.072   

   LandMaxTemperature  LandMaxTemperatureUncertainty  LandMinTemperature  \
0                 NaN                            NaN                 NaN   
1                 NaN                            NaN                 NaN   
2                 NaN                            NaN                 NaN   
3                 NaN                            NaN                 NaN   
4                 NaN                            NaN                 NaN   

   LandMinTemperatureUncertainty  LandAndOceanAverageTemperature  \
0                            NaN                             NaN   
1                            NaN                             NaN   
2                            NaN                             NaN   
3                            NaN                             NaN   
4                            NaN                             NaN   

   LandAndOceanAverageTemperatureUncertainty years  
0                                        NaN  1750  
1                                        NaN  1750  
2                                        NaN  1750  
3                                        NaN  1750  
4                                        NaN  1750

In [25]:
# Grouping the data by years and aggregating 2 columns that we want to work on
data = global_temp.groupby('years').agg({'LandAverageTemperature':'mean',
                                  'LandAverageTemperatureUncertainty':'mean'}).reset_index()
data.head()

years  LandAverageTemperature  LandAverageTemperatureUncertainty
0  1750                8.719364                           2.637818
1  1751                7.976143                           2.781143
2  1752                5.779833                           2.977000
3  1753                8.388083                           3.176000
4  1754                8.469333                           3.494250

In [26]:
data['Uncertainity Top'] = data['LandAverageTemperature'] + \
    data['LandAverageTemperatureUncertainty']

data['Uncertainity Bottom'] = data['LandAverageTemperature'] - \
    data['LandAverageTemperatureUncertainty']

In [27]:
data.head()

years  LandAverageTemperature  LandAverageTemperatureUncertainty  \
0  1750                8.719364                           2.637818   
1  1751                7.976143                           2.781143   
2  1752                5.779833                           2.977000   
3  1753                8.388083                           3.176000   
4  1754                8.469333                           3.494250   

   Uncertainity Top  Uncertainity Bottom  
0         11.357182             6.081545  
1         10.757286             5.195000  
2          8.756833             2.802833  
3         11.564083             5.212083  
4         11.963583             4.975083

In [28]:
data.columns

Index(['years', 'LandAverageTemperature', 'LandAverageTemperatureUncertainty',
       'Uncertainity Top', 'Uncertainity Bottom'],
      dtype='object')

In [29]:
px.line(
    data_frame=data,
    x='years',
#    y=['LandAverageTemperature', 'Uncertainity Top', 'Uncertainity Bottom'],
    y=['LandAverageTemperature'],    
    render_mode='auto',
    title='Average Land Temperature in the World',
    width=800,
    height=450,
)

From the above plot, we can see that the LandAverageTemperature has gone up in the last 50 years or so. The upward trend looks like starting from around 1970 and shows a gradual increase in the temperature. Our analysis clearly shows that there is global warming in the world.

### Analysis of average temperature in each season

In [30]:
global_temp.head()

dt  LandAverageTemperature  LandAverageTemperatureUncertainty  \
0  1750-01-01                   3.034                              3.574   
1  1750-02-01                   3.083                              3.702   
2  1750-03-01                   5.626                              3.076   
3  1750-04-01                   8.490                              2.451   
4  1750-05-01                  11.573                              2.072   

   LandMaxTemperature  LandMaxTemperatureUncertainty  LandMinTemperature  \
0                 NaN                            NaN                 NaN   
1                 NaN                            NaN                 NaN   
2                 NaN                            NaN                 NaN   
3                 NaN                            NaN                 NaN   
4                 NaN                            NaN                 NaN   

   LandMinTemperatureUncertainty  LandAndOceanAverageTemperature  \
0                            NaN                             NaN   
1                            NaN                             NaN   
2                            NaN                             NaN   
3                            NaN                             NaN   
4                            NaN                             NaN   

   LandAndOceanAverageTemperatureUncertainty years  
0                                        NaN  1750  
1                                        NaN  1750  
2                                        NaN  1750  
3                                        NaN  1750  
4                                        NaN  1750

We do not have any date on season so will have to separate month from date and work for season analysis using month info.

In [31]:
global_temp['dt'].dtype

dtype('O')

In [32]:
# Convert date to datetime format
global_temp['dt'] = pd.to_datetime(global_temp['dt'])

In [33]:
# Separating month from date
global_temp['month'] = global_temp['dt'].dt.month

In [34]:
global_temp.head()

dt  LandAverageTemperature  LandAverageTemperatureUncertainty  \
0 1750-01-01                   3.034                              3.574   
1 1750-02-01                   3.083                              3.702   
2 1750-03-01                   5.626                              3.076   
3 1750-04-01                   8.490                              2.451   
4 1750-05-01                  11.573                              2.072   

   LandMaxTemperature  LandMaxTemperatureUncertainty  LandMinTemperature  \
0                 NaN                            NaN                 NaN   
1                 NaN                            NaN                 NaN   
2                 NaN                            NaN                 NaN   
3                 NaN                            NaN                 NaN   
4                 NaN                            NaN                 NaN   

   LandMinTemperatureUncertainty  LandAndOceanAverageTemperature  \
0                            NaN                             NaN   
1                            NaN                             NaN   
2                            NaN                             NaN   
3                            NaN                             NaN   
4                            NaN                             NaN   

   LandAndOceanAverageTemperatureUncertainty years  month  
0                                        NaN  1750      1  
1                                        NaN  1750      2  
2                                        NaN  1750      3  
3                                        NaN  1750      4  
4                                        NaN  1750      5

In [35]:
# Function that defines seasons from months of a year
def get_season(month):
    if month >= 3 and month <= 5:
        return 'spring'
    elif month >= 6 and month <= 8:
        return 'summer'
    elif month >= 9 and month <= 11:
        return 'autumn'
    else:
        return 'winter'

In [36]:
# applying the get_season function to the month column
global_temp['season'] = global_temp['month'].apply(get_season)

In [37]:
global_temp.head()

dt  LandAverageTemperature  LandAverageTemperatureUncertainty  \
0 1750-01-01                   3.034                              3.574   
1 1750-02-01                   3.083                              3.702   
2 1750-03-01                   5.626                              3.076   
3 1750-04-01                   8.490                              2.451   
4 1750-05-01                  11.573                              2.072   

   LandMaxTemperature  LandMaxTemperatureUncertainty  LandMinTemperature  \
0                 NaN                            NaN                 NaN   
1                 NaN                            NaN                 NaN   
2                 NaN                            NaN                 NaN   
3                 NaN                            NaN                 NaN   
4                 NaN                            NaN                 NaN   

   LandMinTemperatureUncertainty  LandAndOceanAverageTemperature  \
0                            NaN                             NaN   
1                            NaN                             NaN   
2                            NaN                             NaN   
3                            NaN                             NaN   
4                            NaN                             NaN   

   LandAndOceanAverageTemperatureUncertainty years  month  season  
0                                        NaN  1750      1  winter  
1                                        NaN  1750      2  winter  
2                                        NaN  1750      3  spring  
3                                        NaN  1750      4  spring  
4                                        NaN  1750      5  spring

In [38]:
# Unique years in the data
years = global_temp['years'].unique()

In [39]:
# Temp lists to store season temperatures in each year
spring_temps = []
summer_temps = []
autumn_temps = []
winter_temps = []

In [40]:
# Find average temperature of each season in each year
for year in years:
    current_df = global_temp[global_temp['years'] == year]
    spring_temps.append(current_df[current_df['season'] == 'spring']['LandAverageTemperature'].mean())
    summer_temps.append(current_df[current_df['season'] == 'summer']['LandAverageTemperature'].mean())
    autumn_temps.append(current_df[current_df['season'] == 'autumn']['LandAverageTemperature'].mean())
    winter_temps.append(current_df[current_df['season'] == 'winter']['LandAverageTemperature'].mean())

In [41]:
spring_temps

[8.563,
 6.734999999999999,
 7.035499999999999,
 8.627333333333334,
 9.074333333333334,
 8.583666666666666,
 9.466,
 8.604666666666667,
 6.896666666666666,
 6.897333333333333,
 6.653666666666666,
 8.916,
 7.809333333333332,
 6.716,
 8.192,
 8.868666666666668,
 8.432333333333332,
 7.831,
 6.144000000000001,
 8.803333333333333,
 7.132000000000001,
 6.0523333333333325,
 7.148666666666666,
 8.866999999999999,
 10.607,
 9.036666666666667,
 7.522333333333333,
 7.774333333333334,
 8.957999999999998,
 10.370666666666667,
 11.737666666666664,
 7.599,
 7.390999999999998,
 8.397333333333334,
 7.3580000000000005,
 6.173666666666667,
 8.849666666666666,
 7.9576666666666656,
 8.159333333333334,
 7.783,
 6.997333333333333,
 7.9253333333333345,
 7.914666666666666,
 8.248,
 9.146333333333333,
 8.552,
 7.507666666666666,
 7.024333333333334,
 8.953333333333333,
 8.041666666666666,
 8.224666666666666,
 8.660666666666666,
 7.760333333333333,
 8.653666666666666,
 8.863,
 8.328999999999999,
 8.07533333333333

In [42]:
# Create a season dataframe for season temperature of each year
season = pd.DataFrame()

In [43]:
# Populting season dataframe with the year and season temperature info
season['year'] = years
season['spring_temps'] = spring_temps
season['summer_temps'] = summer_temps
season['autumn_temps'] = autumn_temps
season['winter_temps'] = winter_temps

In [44]:
season.head()

year  spring_temps  summer_temps  autumn_temps  winter_temps
0  1750      8.563000     14.518333      8.890000      2.963000
1  1751      6.735000     14.116000     10.673000      1.729000
2  1752      7.035500           NaN      7.587000      2.717000
3  1753      8.627333     14.608333      9.212333      1.104333
4  1754      9.074333     14.208333      8.957333      1.637333

In [45]:
season.columns

Index(['year', 'spring_temps', 'summer_temps', 'autumn_temps', 'winter_temps'], dtype='object')

In [46]:
# Visualizing the seasonal temperature trend
fig = px.line(
    data_frame=season,
    x='year',
    y=['spring_temps', 'summer_temps', 'autumn_temps', 'winter_temps'],
    render_mode='auto',
    title='Average Temperature in Each Season',
    width=None,
    height=None,
)
fig.show()

From the above plot, we can infer that the temperature in each season has started to go up from 1970 and we can conclude that the rise in temperature is due to global warming.

### Analysis of trend in temperatures for the top economies

In [47]:
# Creating a list of top economies
continent = ['Russia','United States','China','Japan','Australia','India']

In [48]:
global_temp_country.head()

dt  AverageTemperature  AverageTemperatureUncertainty Country
0  1743-11-01               4.384                          2.294   Åland
5  1744-04-01               1.530                          4.680   Åland
6  1744-05-01               6.702                          1.789   Åland
7  1744-06-01              11.609                          1.577   Åland
8  1744-07-01              15.342                          1.410   Åland

In [49]:
# separating top economies from global_temp_country dataframe
continent_df = global_temp_country[global_temp_country['Country'].isin(continent)]
continent_df.head()

dt  AverageTemperature  AverageTemperatureUncertainty  \
34816  1852-07-01              14.116                          1.530   
34817  1852-08-01              15.330                          1.400   
34818  1852-09-01              18.740                          1.446   
34819  1852-10-01              21.984                          1.493   
34820  1852-11-01              24.073                          1.466   

         Country  
34816  Australia  
34817  Australia  
34818  Australia  
34819  Australia  
34820  Australia

In [50]:
# Applying fetch_year function to to get the year from dt column
continent_df['years'] = continent_df['dt'].apply(fetch_year)

C:\Users\unpat\AppData\Local\Temp/ipykernel_27756/2516659634.py:2: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [51]:
# avg_temp = continent_df.groupby(['Country','years']).agg({'AverageTemperature':'mean'}).reset_index()
avg_temp = continent_df.groupby(['years','Country']).agg({'AverageTemperature':'mean'}).reset_index()
avg_temp.head()

years        Country  AverageTemperature
0  1768  United States            5.572750
1  1769  United States           10.446500
2  1774  United States            1.603000
3  1775  United States            9.499167
4  1776  United States            8.110000

In [126]:
avg_temp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1148 entries, 0 to 1147
Data columns (total 3 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   years               1148 non-null   object 
 1   Country             1148 non-null   object 
 2   AverageTemperature  1148 non-null   float64
dtypes: float64(1), object(2)
memory usage: 27.0+ KB


We currently have years as object so it should be converted to int before we plot a graph

In [127]:
avg_temp['years'] = avg_temp['years'].astype(int)
avg_temp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1148 entries, 0 to 1147
Data columns (total 3 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   years               1148 non-null   int32  
 1   Country             1148 non-null   object 
 2   AverageTemperature  1148 non-null   float64
dtypes: float64(1), int32(1), object(1)
memory usage: 22.5+ KB


In [128]:
fig = px.line(avg_temp, x = 'years', y = 'AverageTemperature', color = 'Country', 
        title = 'Average Land Temperature in the Top Economies')
fig.show()

We can infer that the temperature started to rise in these top economies from around 1970.

In [149]:
# Average Land Temperature in the Top Economies between 1920 to 1970 and 1970 to 2013 for further analysis
fig1 = px.line(avg_temp[600:880], x = 'years', y = 'AverageTemperature', color = 'Country', 
        title = 'Average Land Temperature in the Top Economies - 1920 ~ 1970')
fig2 = px.line(avg_temp[880:], x = 'years', y = 'AverageTemperature', color = 'Country', 
        title = 'Average Land Temperature in the Top Economies - 1970 ~ 2013')

fig1.show()
fig2.show()

For a comparison from the above plots, there is some increase in the temperature during the last 50 years while there is not much increase during the previous 50 years.

### Analysis of average temperatures of US states